src/ast/irbuilderbpf.cpp

```diff
@@ -348,6 +348,12 @@ Value *IRBuilderBPF::CreateUSDTReadArgument(Value *ctx, struct bcc_usdt_argument
   if (argument->valid & BCC_USDT_ARGUMENT_BASE_REGISTER_NAME) {
     int offset = 0;
     offset = arch::offset(argument->base_register_name);
+    if (offset < 0) {
```

Comment:
Doesn't the semantic analyser catch this? Semantic analyser gives nicer errors than this here does.

Reply:
No, it silently uses offset -1. I don't think the semantic analyser sees `base_register_name`: it comes from libbcc in the second overload of `CreateUSDTReadArgument` just below.

Reply:
We can probably move it, but it will be quite a lot of work for a slightly nicer error message